File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task13_PERPLEXITY_llama-3-sonar-large-32k-chat.c
#include <stdio.h>
#include <string.h>

int isValidWalk(char* walk[], int len) {
    int x = 0, y = 0;
    for (int i = 0; i < len; i++) {
        if (strcmp(walk[i], "n") == 0) y++;
        if (strcmp(walk[i], "s") == 0) y--;
        if (strcmp(walk[i], "e") == 0) x++;
        if (strcmp(walk[i], "w") == 0) x--;
    }
    return x == 0 && y == 0 && len == 10;
}

int main() {
    char* walk[] = {"n", "s", "e", "w", "n", "s", "e", "w", "n", "s"};
    int len = sizeof(walk) / sizeof(walk[0]);
    printf("%d
", isValidWalk(walk, len));
    return 0;
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task13_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Main {
    public static boolean isValidWalk(String[] walk) {
        int x = 0, y = 0;
        for (String dir : walk) {
            if (dir.equals("n")) y++;
            if (dir.equals("s")) y--;
            if (dir.equals("e")) x++;
            if (dir.equals("w")) x--;
        }
        return x == 0 && y == 0 && walk.length == 10;
    }

    public static void main(String[] args) {
        String[] walk = {"n", "s", "e", "w", "n", "s", "e", "w", "n", "s"};
        System.out.println(isValidWalk(walk));
    }
}